borrowing/models.py

```diff
@@ -18,3 +21,32 @@ class Borrowing(models.Model):
         on_delete=models.CASCADE,
         related_name="borrowings",
     )
+
+    def __str__(self) -> str:
+        return (
+            f"Book: {self.book.title}, Author: {self.book.author}. "
+            f"Borrowed from {self.borrow_date} to {self.expected_return_date}."
+        )
+
+    def clean(self):
+        if not self.borrow_date:
+            self.borrow_date = date.today()
+
+        if self.expected_return_date and self.expected_return_date < self.borrow_date:
+            raise ValidationError(
+                "Expected return date cannot be set up earlier than the borrow date."
+            )
+
+        if self.actual_return_date and self.actual_return_date < self.borrow_date:
+            raise ValidationError(
+                "Expected return date cannot be earlier than the borrow date."
+            )
+
+        if self.book.inventory == 0:
```

Comment:
Instead of checking if inventory is zero, you should check if inventory is less than or equal to zero to account for cases where the number of books is exhausted